### ISSUE TITLE
MODULE.bazel.lock update failure when bumping Stardoc dep from 0.7.0 to 0.7.1

### ISSUE THREAD

**Issue by calebzulawski** [bazelbuild/stardoc#257]
I am upgrading stardoc from 0.7.0 to 0.7.1 and get the following warning when updating MODULE.bazel.lock:
```
$ bazel mod deps --lockfile_mode=update
WARNING: The module extension @@rules_jvm_external~//:extensions.bzl%maven produced an invalid lockfile entry because it referenced @@[unknown repo '' requested from @@rules_jvm_external~]. Please report this issue to its maintainers.
```
MODULE.bazel.lock ends up broken:
```
$ bazel mod deps --lockfile_mode=error
ERROR: MODULE.bazel.lock is no longer up-to-date because: The repo mappings of certain repos used by the extension 'ModuleExtensionId{bzlFileLabel=@@rules_jvm_external~//:extensions.bzl, extensionName=maven, isolationKey=Optional.empty}' have changed. Please run `bazel mod deps --lockfile_mode=update` to update your lockfile. Type 'bazel help mod' for syntax and help.
```
I am using Bazel 7.4.0. Thanks!

**Comment by tetromino** [comment]
I can reproduce the failure:

```bash
mkdir x
cd x
echo 'bazel_dep(name = "stardoc", version = "0.7.0")' > MODULE.bazel
USE_BAZEL_VERSION=7.4.0 bazelisk mod deps
sed -e 's/0.7.0/0.7.1/' -i MODULE.bazel
USE_BAZEL_VERSION=7.4.0 bazelisk mod deps --lockfile_mode=update
USE_BAZEL_VERSION=7.4.0 bazelisk mod deps --lockfile_mode=error
```

Result: `mod deps --lockfile_mode=update` prints `WARNING: The module extension @@rules_jvm_external~//:extensions.bzl%maven produced an invalid lockfile entry because it referenced @@[unknown repo '' requested from @@rules_jvm_external~]. Please report this issue to its maintainers.`, and `mod deps --lockfile_mode=error` fails.

@Wyverald - any suggestions for what is happening here?

**Comment by tetromino** [comment]
This breakage appears only with Bazel 7.4.0 (reproducer works fine in Bazel 7.3.2)

**Comment by fmeum** [comment]
This is ultimately an issue in r_j_e that should be fixed by https://github.com/bazel-contrib/rules_jvm_external/pull/1265. The warning comes from https://github.com/bazelbuild/bazel/commit/8cee5f5e6aca1e553a4434c9e7ef4bafd3e723c3. I recall that it didn't result in failures when I tried it on r_j_e a while back, but that may have changed in the meantime.

**Comment by calebzulawski** [comment]
Thank you for looking into this!  I'll hold off upgrading for now.

**Comment by chickenandpork** [comment]
I ran into this problem; I found that putting a redundant dependency in my `MODULE.bazel` fixed:
```
bazel_dep(name = "rules_jvm_external", version = "6.2")
```

I don't *need* `rules_jvm_external` directly, but by placing this dependency, it overrides the `6.1` in stardoc's `MODULE.bazel` file:

https://github.com/bazelbuild/stardoc/blob/46fd77a446f00faddedd0467ce993b462fba3c8a/MODULE.bazel#L10
```
bazel_dep(name = "rules_java", version = "7.6.1")
bazel_dep(name = "rules_jvm_external", version = "6.1")  <--- 
bazel_dep(name = "rules_license", version = "0.0.7")
```

Really, version 6.5 is available, but I found you really need to get off `6.1`.  Something with rules_jvm_external-6.1 fights with Bazel-7.4

If this release that mitigates is almost shipping, it's probably too late, but if you're stuck, claiming an unnecessary direct dependency unblocked me.

**Comment by tetromino** [comment]
I beleive that #263 fixed this (at least when using Bazel 7.4.1)
